openshift-kube-apiserver/handlers/deprecatedapirequest/handler.go

```diff
@@ -0,0 +1,30 @@
+package deprecatedapirequest
+
+import (
+	"net/http"
+	"time"
+
+	apiclient "github.com/openshift/client-go/apiserver/clientset/versioned"
+	apiinformers "github.com/openshift/client-go/apiserver/informers/externalversions"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/client-go/rest"
+)
+
+// WithDeprecatedApiRequestHandler adds an http handler that logs requests to deprecated apis.
+func WithDeprecatedApiRequestHandler(handler http.Handler, config *genericapiserver.Config) http.Handler {
+	clientConfig := rest.CopyConfig(config.LoopbackClientConfig)
+	clientConfig.AcceptContentTypes = "application/json"
+	clientConfig.ContentType = "application/json"
+	client, err := apiclient.NewForConfig(clientConfig)
+	if err != nil {
+		runtime.HandleError(err)
+		return handler
+	}
+	informers := apiinformers.NewSharedInformerFactory(client, 10*time.Minute)
+	lister := informers.Apiserver().V1().DeprecatedAPIRequests().Lister()
```

Comment:
who starts these? Usually we do the plumbing for informers at the very outer level.

Reply:
The `initializeInformers` filter added to the chain below starts the informer.